exp/runtime/hooks/api/v1alpha1/lifecyclehooks_types.go

```diff
@@ -28,6 +28,9 @@ import (
 type BeforeClusterCreateRequest struct {
 	metav1.TypeMeta `json:",inline"`
 
+	// CommonRequest contains Settings field common to all request types.
```

Comment:
What about keeping this generic so it won't become outdated if we add more fields to CommonRequest, e.g
```suggestion
	// CommonRequest contains fields common to all request types.
```